ledger/participant-state/src/main/scala/com/daml/ledger/participant/state/v2/AdaptedV1ReadService.scala

```diff
@@ -0,0 +1,178 @@
+// Copyright (c) 2021 Digital Asset (Switzerland) GmbH and/or its affiliates. All rights reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package com.daml.ledger.participant.state.v2
+
+import java.util.UUID
+
+import akka.NotUsed
+import akka.stream.scaladsl.Source
+import com.daml.ledger.api.health.HealthStatus
+import com.daml.ledger.participant.state.v1
+import com.daml.ledger.participant.state.v2.AdaptedV1ReadService._
+import com.daml.ledger.participant.state.v2.Update.CommandRejected
+import com.daml.ledger.participant.state.v2.Update.CommandRejected.RejectionReasonTemplate
+
+class AdaptedV1ReadService(delegate: v1.ReadService) extends ReadService {
```

Comment:
AFAICS this read service does not ensure the deduplication guarantee of the `v2.ReadService` using only the assumptions on a `v1.ReadService`. Do you want to add a disclaimer to this class?

Reply:
Added in #10254.